web3.js/src/message/versioned.ts

```diff
@@ -5,17 +5,26 @@ import {MessageV0} from './v0';
 export type VersionedMessage = Message | MessageV0;
 // eslint-disable-next-line no-redeclare
 export const VersionedMessage = {
```

Comment:
I haven't checked how tree-shakers treat this. If I do:

```
import {VersionedMessage} from 'src/message/versioned';
// ...but then only use one function
if (VersionedMessage.deserializeMessageVersion(message) === 'legacy') {
  /* ... */
}
```

…does the compiler shake out the other methods I haven't used?

I know it definitely works if you export each function.

```
export function deserializeMessageVersion(...) { ... }
```

…then

```
import {deserializeMessageVersion} from 'src/message/versioned';
if (deserializeMessageVersion(message) === 'legacy') {
  /* ... */
}
```

Reply:
Oh good question, I'd like to punt on this for now and would happily accept / implement a change that's more tree-shaker friendly if it's an issue